### PR TITLE
BEA: add county/sector/year GDP from bea

### PR DESCRIPTION
**Pull request by sglyon**

### CODE REVIEW

File: src/cmdc/datasets/bea/data.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+from concurrent.futures import ThreadPoolExecutor
+import textwrap
+import requests
+import pandas as pd
+
+from cmdc.datasets.base import OnConflictNothingBase
+
+
+def _make_bea_request(**kw):
+    key = "94DEC0DC-E6F5-4DC9-8E78-64D61AA34433"
```
Comment (cc7768) [inline src/cmdc/datasets/bea/data.py:11]: We probably don't want this key to be public info.

File: src/cmdc/datasets/bea/data.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+
+class CountyGDP(OnConflictNothingBase):
+    pk = '("id", "year", "fips")'
+    table_name = "bea_gdp"
+
+    def __init__(self, year):
+        super().__init__()
+        self.year = year
+
```
Comment (cc7768) [inline src/cmdc/datasets/bea/data.py:58]: Double new line

File: src/cmdc/datasets/bea/data.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        out = f"""
+        INSERT INTO data.{table_name} (id, year, fips, value)
+        SELECT bv.id, year, fips, value
+        from {temp_name} tt
+        LEFT JOIN meta.bea_variables bv using (dataset, tablename, line_code)
+        INNER JOIN data.us_counties using(fips)
+        ON CONFLICT {pk} DO UPDATE set value = excluded.value;
+        """
+        return textwrap.dedent(out)
+
```
Comment (cc7768) [inline src/cmdc/datasets/bea/data.py:69]: Double new line

**Review by cc7768 — CHANGES_REQUESTED** [review]
Looks close.

You have a method to get the variables from the bea api and put them into a DataFrame but I don't see how they're pushed into the SQL database. Are you just doing this manually?

**Comment by sglyon** [comment]
> You have a method to get the variables from the bea api and put them into a DataFrame but I don't see how they're pushed into the SQL database. Are you just doing this manually?

Yes, I just did this by hand -- good catch.

File: src/cmdc/datasets/bea/data.py (hunk 1 above)
Comment (sglyon) [reply src/cmdc/datasets/bea/data.py:11]: Yeah... I'll change it to read from env var

**Pull request merged**